lib/ansible/modules/cloud/vmware/vmware_guest_move.py

```diff
@@ -0,0 +1,166 @@
+        default: 'first'
+        choices: [ first, last ]
+   folder:
+        description:
+            - Absolute path to move an existing guest
+            - The folder should include the datacenter. ESX's datacenter is ha-datacenter.
+            - This parameter is case sensitive.
+            - 'If multiple machines are found with same name, this parameter is required to identify
+              uniqueness of the virtual machine.'
+            - 'Examples:'
+            - '   folder: /ha-datacenter/vm'
+            - '   folder: ha-datacenter/vm'
+            - '   folder: /datacenter1/vm'
+            - '   folder: datacenter1/vm'
+            - '   folder: /datacenter1/vm/folder1'
+            - '   folder: datacenter1/vm/folder1'
+            - '   folder: /folder1/datacenter1/vm'
+            - '   folder: folder1/datacenter1/vm'
+            - '   folder: /folder1/datacenter1/vm/folder2'
+        default: /vm
+   datacenter:
+        description:
+            - Destination datacenter for the deploy operation
+        required: True
+extends_documentation_fragment: vmware.documentation
+'''
+
+EXAMPLES = r'''
+- name: Move Virtual Machine
+  vmware_guest_move:
+    hostname: "{{ vcenter_ip }}"
+    username: "{{ vcenter_username }}"
+    password: "{{ vcenter_password }}"
+    datacenter: datacenter
+    validate_certs: False
+    name: testvm-1
+    folder: datacenter/vm/prodvms
+
+
+- name: Move Virtual Machine - UUID
+  vmware_guest_move:
+    hostname: "{{ vcenter_ip }}"
+    username: "{{ vcenter_username }}"
+    password: "{{ vcenter_password }}"
+    datacenter: datacenter
+    validate_certs: False
+    uuid: "{{ vm_uuid }}"
+    folder: datacenter/vm/testvms
+'''
+
+RETURN = """
+instance:
+    description: metadata about the virtual machine
+    returned: always
+    type: dict
+    sample: None
+"""
+
+try:
+    import pyVmomi
+    from pyVmomi import vim
+except ImportError:
+    pass
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_text
+from ansible.module_utils.vmware import PyVmomi, vmware_argument_spec, connect_to_api
+
+
+class PyVmomiHelper(PyVmomi):
+    def __init__(self, module):
+        super(PyVmomiHelper, self).__init__(module)
+
+
+def main():
+    argument_spec = vmware_argument_spec()
+    argument_spec.update(
+        name=dict(type='str'),
+        name_match=dict(
+            type='str', choices=['first', 'last'], default='first'),
+        uuid=dict(type='str'),
+        folder=dict(type='str', default='/vm'),
+        datacenter=dict(type='str', required=True),
+    )
+    module = AnsibleModule(
+        argument_spec=argument_spec, required_one_of=[['name', 'uuid']])
+
+    # FindByInventoryPath() does not require an absolute path
+    # so we should leave the input folder path unmodified
+    module.params['folder'] = module.params['folder'].rstrip('/')
+    content = connect_to_api(module)
```

Comment:
no need for this as `pyv.content` will return same.

```
search_index = pyv.content.searchIndex
```

Reply:
Changed also the order pyv is called and two lines below.